projects/bootloader/scripts/can_datagram.py

```diff
@@ -0,0 +1,294 @@
+
+        # Verify all inputs
+        assert kwargs["datagram_type_id"] & 0xff == kwargs["datagram_type_id"]
+
+    def _convert_to_bytearray(self, in_value, bytes):
+        """This is a helper function that creates a little-endian bytearray"""
+        out_bytearray = bytearray()
+        for i in range(0, bytes):
+            out_bytearray.append((in_value >> (8 * i)) & 0xff)
+        return out_bytearray
+
+    def _convert_from_bytearray(self, in_bytearray, bytes):
+        """This is a helper function that converts a bytearray into a value"""
+        value = 0
+        for i in range(0, bytes):
+            value = value | ((in_bytearray[i] & 0xff) << (i * 8))
+        return value
+
+    def _calculate_crc32(self, datagram_type_id, node_ids, data):
+        """This function returns a bytearray based on set data."""
+
+        node_crc32 = zlib.crc32(bytearray(node_ids))
+        node_crc32 = self._convert_to_bytearray(self, node_crc32, 4)
+        data_crc32 = zlib.crc32(data)
+        data_crc32 = self._convert_to_bytearray(self, data_crc32, 4)
+
+        crc32_array = bytearray([datagram_type_id,
+                                 len(node_ids),
+                                 *node_crc32,
+                                 len(data) & 0xff,
+                                 (len(data) >> 8) & 0xff,
+                                 *data_crc32])
+        # Update the crc32
+        crc32 = zlib.crc32(crc32_array)
+        crc32 = self._convert_to_bytearray(self, crc32, 4)
+
+        # Update the bytearray
+        return crc32
+
+
+class DatagramSender:
+    """A class that acts as a distributor for the Datagram class on a bus."""
+
+    def __init__(self, bustype="socketcan", channel=DEFAULT_CHANNEL,
+                 bitrate=CAN_BITRATE, receive_own_messages=False):
+        self.bus = can.interface.Bus(
+            bustype=bustype,
+            channel=channel,
+            bitrate=bitrate,
+            receive_own_messages=receive_own_messages)
+
+    def send(self, message):
+        """Sends the Datagrams."""
+
+        assert isinstance(message, Datagram)
+
+        chunk_messages = self._chunkify(message.serialize(), 8)
+        message_arbitration_id = CAN_ARBITRATION_ID
+        message_extended_arbitration = False
+
+        can_messages = [
+            can.Message(
+                arbitration_id=CAN_START_ARBITRATION_ID,
+                data=bytearray(),
+                is_extended_id=message_extended_arbitration)]
+
+        # Populate an array with the can message from the library
+        for chunk_message in chunk_messages:
+            can_messages.append(can.Message(arbitration_id=message_arbitration_id,
+                                            data=chunk_message,
+                                            is_extended_id=message_extended_arbitration))
+
+            # After the first message, set the arbitration ID to 0
+            message_arbitration_id = CAN_ARBITRATION_ID
```

Comment:
I don't think you need these lines anymore since `message_arbitration_id` never changes